nvm.sh

```diff
@@ -299,6 +299,530 @@ nvm_find_up() {
   nvm_echo "${path_}"
 }
 
+# NOTE: this function only validates across one line
+nvm_string_contains_regexp() {
+  local string
+  string="${1-}"
+  local regexp
+  regexp="${2-}"
+  if [ -z  "$string" ] || [ -z "$regexp" ]; then
+    return 1
+  fi
+  # e.g. "nvm_string_contains_regexp abbc ^aa?b+.$" returns 0
+  command printf "%s" "$string" | command awk "/$regexp/{ exit 0 }{ exit 1 }"
+}
+
+# Validates that the given semver adheres to the following grammar:
+#
+# semver         ::= comparator_set ( ' || '  comparator_set )*
+# comparator_set ::= comparator ( ' ' comparator )*
+# comparator     ::= ( '<' | '<=' | '>' | '>=' | '' ) [0-9]+ '.' [0-9]+ '.' [0-9]+
+nvm_is_valid_semver() {
+  if nvm_string_contains_regexp "${1-}" '^( ?(<|<=|>|>=)?[0-9]+\.[0-9]+\.[0-9]+)+( \|\| ( ?(<|<=|>|>=)?[0-9]+\.[0-9]+\.[0-9]+)+)*$'; then
```

Comment:
could this function just be the `nvm_string_contains_regexp` call, without the if/else and return 0/1?